zephyr-make

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/make -rf
+
+BOARD ?= qemu_x86
+CONF_FILE = prj.conf
+
+TARGETS = run flash debug debugserver ram_report rom_report
+
+all: outdir/${BOARD}/zephyr.elf
+
+outdir/${BOARD}/Makefile: CMakeLists.txt
+	mkdir -p outdir/${BOARD} && cmake -DCONF_FILE=${CONF_FILE} -DBOARD=${BOARD} -Boutdir/${BOARD} -H.
+
```

Comment:
This does not support incremental builds, arguably the most important use-case of a build system
since it is invoked so often by users.

For example, editing a C source file, and then running "make" will not trigger a rebuild because
(AFAICT from just reading this) this Makefile is not aware that the elf file depends on C source files.

Maybe you could fix this with some PHONY targets, not sure.